services/QuillLMS/engines/evidence/lib/evidence/evidence/check.rb

```diff
@@ -59,5 +61,16 @@ def self.fallback_feedback
 
       FALLBACK_RESPONSE
     end
+
+    def self.checks_to_run(feedback_types)
+      return ALL_CHECKS if feedback_types.nil? || feedback_types.empty?
+
+      qualified_feedback_types = feedback_types.map { |t| "Evidence::Check::#{t}" }
+      filtered_checks = ALL_CHECKS.select { |check| qualified_feedback_types.include?(check.name) }
+
+      raise NoMatchedFeedbackTypesError if filtered_checks.empty?
```

Comment:
This changes the business logic beyond the scope of the Notion spec. I suggest sticking with the original behavior of returning FALLBACK_FEEDBACK rather than raise an error in this case.  

If there's a good reason to fail here, let's mention it in the spec, and raise it earlier at the controller params level.

Reply:
We could do that, but returning default feedback will, generally speaking, make it impossible for you to tell the difference between "the response is optimal for the specified algorithms" and "the algorithms specified are invalid".  You'll get exactly the same response in both cases.  That ambiguity strikes me as a pretty bad outcome.

I'm also not clear on what the risk of 500-ing sometimes is, but maybe I'm missing something obvious?

Reply:
<img width="1180" alt="Screen Shot 2022-06-15 at 2 11 58 PM" src="https://user-images.githubusercontent.com/90669/173918300-1c1c3075-fced-4876-b4b3-3148fa9fdf02.png">

If a user ever got into this state, the UX would fail hard, for example. The evidence flow doesn't currently handle API failures gracefully.

Reply:
I agree with your concern about different errors getting compressed into FALLBACK_FEEDBACK in a confusing way, so that should be addressed one way or another.

Reply:
I'm torn.  On the one hand, you're right, the front-end can't really handle errors well.  On the other, would we ever want to use this for anything other than Postman-style API testing?  That is, would the front-end ever send the params necessary to trigger this code branch?  I suppose it's dangerous to assert that something will "never" happen in the future.

Maybe we should just send the error via `ErrorNotifier` and then swallow it?  The downside there is that if you're just looking at HTTP responses, you still can't tell the difference, and need to be in Sentry and/or NewRelic, but perhaps that's acceptable?

Reply:
We could avoid a hard error, and also improve messaging, but having the FALLBACK_FEEDBACK display different texts, depending on the code path of the error. Just an idea